phip/hit_calling.py

```diff
@@ -0,0 +1,285 @@
+import time
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import minimize_scalar
+
+
+def do_hit_calling(
+        counts_df,
+        beads_only_samples,
+        fdr=0.15,
+        smoothing_bracket=(0, 500),
+        verbosity=2):
+    """
+    Call hits at the specified FDR using a heuristic.
+
+    A hit is a (clone, sample) pair whose read count exceeds what is expected
+    by chance.
+
+    The basic assumption is that the beads-only (negative control) samples
+    for some clone c give a good indication of the read count distribution
+    of non-hits for that clone. Background effects that are not captured by
+    beads-only samples will not be accounted for by this routine. At least
+    two beads-only samples are required, but more are better.
+
+    Overview of the algorithm:
+
+    An intuitive quantity to consider for a clone c and sample s is the
+    "smoothed ratio" between the read count at entry (c, s) and the maximum
+    number of reads across beads-only samples for c. The smoothed ratio is
+    given by (a + smoothing) / (b + smoothing) for some value of `smoothing`.
```

Comment:
This reminds me of the situation of using a Dirichlet prior. Is there some analog probabilistic model here that could clarify it all?